src/storage/write_ahead_log/log_serializer_task.cpp

```diff
@@ -136,6 +137,16 @@ BufferedLogWriter *LogSerializerTask::GetCurrentWriteBuffer() {
  * Hand over the current buffer and commit callbacks for commit records in that buffer to the log consumer task
  */
 void LogSerializerTask::HandFilledBufferToWriter() {
+  // Mark the buffer as ready for serialization, if it exists. It may not exist for read-only transactions.
+  if (filled_buffer_ != nullptr) filled_buffer_->PrepareForSerialization();
+  // Replicate the buffer, if it exists.
+  if (filled_buffer_ != nullptr) {
+    if (replication_manager_ != DISABLED) {
+      replication_manager_->ReplicateBuffer(filled_buffer_);
+    } else {
+      filled_buffer_->MarkSerialized();
```

Comment:
See my comment in `log_io.h`. If you end up not making that change, I think adding a comment here explaining why we call `MarkSerialized` would be helpful.

Reply:
Removed. It shouldn't be necessary if the Prepare call is setting the right count.